src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -42,6 +43,145 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+            THROW_IF_FAILED(fontFace5->GetFontResource(fontResource.put()));
+
+            std::vector<DWRITE_FONT_AXIS_VALUE> axesVector;
+            const auto axesCount = fontFace5->GetFontAxisValueCount();
+            if (axesCount > 0)
+            {
+                uint32_t localeIndex;
+                BOOL localeExists;
+                wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
+                const auto localeToTry = GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH) ? localeName : L"en-US";
+
+                axesVector.resize(axesCount);
+                fontFace5->GetFontAxisValues(axesVector.data(), axesCount);
+                for (uint32_t i = 0; i < axesCount; ++i)
+                {
+                    const auto tagString = _axisTagToString(axesVector[i].axisTag);
+
+                    wil::com_ptr<IDWriteLocalizedStrings> names;
+                    THROW_IF_FAILED(fontResource->GetAxisNames(i, names.put()));
+
+                    if (!SUCCEEDED(names->FindLocaleName(localeToTry, &localeIndex, &localeExists)) || !localeExists)
+                    {
+                        // default to the first locale in the list
+                        localeIndex = 0;
+                    }
+
+                    uint32_t length;
+                    if (SUCCEEDED(names->GetStringLength(localeIndex, &length)))
+                    {
+                        // it is reasonable to assume that the name length is not going to exceed 512 chars
+                        wchar_t name[512];
+                        if (SUCCEEDED(names->GetString(localeIndex, name, length + 1)))
+                        {
+                            fontAxesTagsAndNames.Insert(tagString, winrt::hstring{ name });
+                            continue;
+                        }
+                    }
+                    // if there was no name found, it means the font does not actually support this axis
+                    // don't insert anything into the vector in this case
+                }
+            }
+        }
+        return fontAxesTagsAndNames;
+    }
+
+    winrt::hstring Font::_axisTagToString(DWRITE_FONT_AXIS_TAG tag)
+    {
+        std::wstring result;
+        for (int i = 0; i < 4; ++i)
+        {
+            result.push_back((tag >> (i * 8)) & 0xFF);
+        }
+        return winrt::hstring{ result };
+    }
+
+    AxisKeyValuePair::AxisKeyValuePair(winrt::hstring axisKey, float axisValue, const Windows::Foundation::Collections::IMap<winrt::hstring, float>& baseMap, const Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring>& tagToNameMap) :
+        _AxisKey{ axisKey },
+        _AxisValue{ axisValue },
+        _baseMap{ baseMap },
+        _tagToNameMap{ tagToNameMap }
+    {
+        if (_tagToNameMap.HasKey(_AxisKey))
+        {
+            int32_t i{ 0 };
+            // this loop assumes that every time we iterate through the map
+            // we get the same ordering
```

Comment:
👀 uh, do we actually know that a `IMap` has that guarantee?

Reply:
BTW do we really need to use an index? `ComboBox` also has `SelectedText` and `SelectedItem` properties so why don't we just use those?

Reply:
> BTW do we really need to use an index?

Unfortunately yes. We cannot just bind `AxisKey` to the ComboBox's `SelectedValue`, `SelectedItem` or even `Text` in a two-way manner, xaml struggles with that for some reason.